src/Stack/Upgrade.hs

```diff
@@ -42,7 +45,16 @@ upgrade gitRepo mresolver = withCanonicalizedSystemTempDirectory "stack-upgrade"
       Just repo -> do
         remote <- liftIO $ readProcess "git" ["ls-remote", repo, "master"] []
         let latestCommit = head . words $ remote
-        if latestCommit == $gitHash then do
+#ifdef USE_GIT_INFO
+            builtHash = $gitHash
+#else
+            builtHash = "git information disabled by flag"
```

Comment:
gitRev sets `$gitHash` to `"UNKNOWN"` in that case, by the way, so you could do the same if !USE_GIT_INFO.